compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -882,14 +902,17 @@ impl CrateInfo {
             let missing =
                 missing.iter().cloned().filter(|&l| lang_items::required(tcx, l)).collect();
             info.missing_lang_items.insert(cnum, missing);
+        }
 
-            // Only include debugger visualizer files from crates that will be statically linked.
-            if used_crate_source.rlib.is_some() || used_crate_source.rmeta.is_some() {
-                let debugger_visualizers = tcx.debugger_visualizers(cnum).clone();
-                if !debugger_visualizers.is_empty() {
-                    info.debugger_visualizers.insert(cnum, debugger_visualizers);
-                }
-            }
+        if tcx.sess.target.is_like_msvc
+            && tcx.sess.crate_types().iter().any(|&crate_type| {
+                crate_type == CrateType::Executable
+                    || crate_type == CrateType::Dylib
+                    || crate_type == CrateType::Cdylib
```

Comment:
For a condition like this it's useful to use an exhaustive match statement, so that we don't miss updating it if a new crate type gets added. It also makes it easy to add documentation for the various cases. E.g.:
```rust 
tcx.sess.crate_types().iter().any(|&crate_type| match {
  CrateType::Executable | CrateType::Dylib | crate_type == CrateType::Cdylib => {
    // These are crate types for which we invoke the linker
    true
  }
  CrateType::ProcMacro => {
    // We could embed NatVis for proc macro crates too (to improve the debugging experience for them)
    // but it does not seem like a good default, since this is a rare use case and we don't want to slow down
    // the common case.
    false
  }
  CrateType::Staticlib | CrateType::Rlib => {
    // We don't invoke the linker for these, so we don't need to collect the NatVis for them.
    false
  }
})
```